thunder/distributed/transforms/fsdp_v2.py

```diff
@@ -116,6 +118,14 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
                             # TODO have a more principled way to update this?
                             a0, _, _, *a2pp = c.args
                             c.args = (a0, tuple(new_shape), thunder_device_str, *a2pp)
+                    # `orig_p` is not None means the input module is initialized with `torch.device("meta")`,
+                    # meaning we should better remove the materialized to free up the unexpected memory.
+                    if orig_p is not None:
+                        orig_p.untyped_storage().resize_(0)
```

Comment:
"the input module is initialized with meta and _then materialized_". I couldn't understand why do we need to free the memory of meta-initialized tensors because the "materialization" step is missing in the comment.